runtime/compiler/z/codegen/UncommonBCDCHKAddressNode.cpp

```diff
@@ -62,17 +62,44 @@ UncommonBCDCHKAddressNode::perform()
             {
             TR::Node* pdOpNode = node->getFirstChild();
             TR::Node* oldAddressNode = node->getSecondChild();
-            TR_ASSERT(pdOpNode && oldAddressNode, "Unexpected null PD opNode or address node under BCDCHK");
-            TR_ASSERT(oldAddressNode->getNumChildren() == 2, "Expecting 2 children under address node of BCDCHK.");
+            TR_ASSERT_FATAL_WITH_NODE(node, pdOpNode && oldAddressNode, "Unexpected null PD opNode or address node under BCDCHK");
 
             TR::ILOpCodes addressOp = oldAddressNode->getOpCodeValue();
-            TR_ASSERT((addressOp == TR::aladd || addressOp == TR::aiadd), "Unexpected addressNode opcode");
+            TR_ASSERT_FATAL_WITH_NODE(node,
```

Comment:
Would it make sense to have the expected tree structure comment moved here before the ASSERT to highlight what you are  checking here?